app/services/catalog/portfolio_permission.rb

```diff
@@ -3,7 +3,7 @@ class PortfolioPermission
     VALID_RESOURCE_VERBS = %w[read write order].freeze
 
     def self.verify_permissions(values)
-      raise Catalog::InvalidParameter, 'Permission should be an array' unless values.kind_of?(Array)
+      raise Catalog::InvalidParameter, 'Permission should be an array' unless values.kind_of?(Array) && values.all? { |value| value.kind_of?(String) } && values.present?
```

Comment:
Possibly move this into a custom validation? Or move it to a mixin with other validations?
In its current form it's a bit long and a challenge to read. Could also be testable and reusable if moved to a method.

Reply:
Sure, I moved this specific check into a `before_action` method which is now in the RBACMixin.